### DISCUSSION TITLE
Convert Dell DW5816e to generic EM7430

### DISCUSSION THREAD

**Discussion by corty8** [danielewood/sierra-wireless-modems#110] [Q&A]
Hi All

Just wondering what the process would be to convert Dell branded EM7430 cards over to generic?

is it possible to use these same tools discussed here to do them?

Cheers
Paul

**Comment by danielewood** [comment]
Tools, yes.

The script is not setup for the proper USB PIDs, so you'll need to change them.  Additionally, it downloads a 7455 specific firmware, so you'll need to use a 7430 firmware instead.

**Comment by corty8** [comment]
Thanks for your reply

Just out of interest is it possible to convert an EM7455 to an EM7430 or at a hardware level they are just not compatible?

I have some EM7455 cards but they do not support the Band 28 which is required here in Aus

**Comment by danielewood** [comment]
Band support is all hardware + low level feature flags.

Different models of the 74xx series support different band sets, even within the 7455 band 30 support is seemingly random.

The 73xx series are an entirely other class of CAT4 (IIRC) modems. The 74xx series are CAT6 modems.

**Reply by corty8** [reply]
Hi Daniel

How are you doing mate?

I have a few of these Dell Cards that I need to convert to generic Sierra which I have been wanting to do for some time now

The script you have written will convert a generic over to dell or Lenovo it appears from memory, if I wanted to go the other way (Dell to Generic) is it just a matter of changing the firmware that it downloads in the script or is there more to it than that?

Cheers

**Comment by iandall** [comment]
I've just done this. I didn't use the script by @danielewood but did rely on the excellent README as a source of information. The core of the matter is to  get the generic firmware from the Sierra Wireless site and update with qmi-firmware-update. In my case though, the modem was in mbim(only) mode and it first had to be put into qmi mode with `qmicli -d /dev/cdc-wdm0 --dms-swi-set-usb-composition=6`. The other trap was initially I only updated the .nvu firmware. You need to update both the .cei and the .nvu to the same version.

I also changed the VID/PID/USBPRODUCT/PRIID but at least for use on linux, I don't think it is strictly necessary. Reasonably recent linux kernels know the Dell VID/PID.

**Reply by 502974830** [reply]
Hello, may I ask if updating the firmware directly can change dw5816e to em7430?
